### PR TITLE
Add Open Graph metadata to the web site pages

### PR DESCRIPTION
**Pull request by trustin**
- Added various Open Graph and Twitter Card metadata.
- Added the default card image.
- Set hand-crafted or auto-generated excerpt to `og:description`
  property.
- Removed short-link pages from the site map.

### CODE REVIEW

**Comment by trustin** [comment]
Deployed this commit to https://armeria.dev/ and it seems to work fine.

File: site/src/pages/index.tsx [hunk 1]
```diff
@@ -73,6 +73,12 @@ const IndexPage: React.FC<RouteComponentProps> = (props) => {
     <BaseLayout
       location={props.location}
       pageTitle="Armeria &ndash; Your go-to microservice framework"
+      pageDescription={
+        'Armeria is your go-to microservice framework for any situation. ' +
+        'You can build any type of microservice leveraging your favorite technologies, ' +
+        'including gRPC, Thrift, Kotlin, Retrofit,Reactive Streams, Spring Boot and Dropwizard. ' +
```
Comment (KarboniteKream) [inline site/src/pages/index.tsx:79]: Missing before `Reactive` 🙂

Comment (trustin) [reply site/src/pages/index.tsx]: Fixed :bow:

**Comment by trustin** [comment]
Thanks, folks! :bow:

**Pull request merged**